File: ToastLibrary/ControlExtensions.cs
﻿using System.ComponentModel;

namespace ToastLibrary;

public static class ControlExtensions
{
    public static void InvokeIfRequired<T>(this T control, Action<T> action) where T : ISynchronizeInvoke
    {
        if (control.InvokeRequired)
        {
            control.Invoke(new Action(() => action(control)), null);
        }
        else
        {
            action(control);
        }
    }
}

File: NotificationsVisualBasic/ControlExtensions.vb
Imports System.ComponentModel

Public Module ControlExtensions
    <Runtime.CompilerServices.Extension>
    Public Sub InvokeIfRequired(Of T As ISynchronizeInvoke)(ByVal control As T, ByVal action As Action(Of T))
        If control.InvokeRequired Then
            control.Invoke(New Action(Sub() action(control)), Nothing)
        Else
            action(control)
        End If
    End Sub
End Module